contrib/win32/win32compat/misc.c

```diff
@@ -909,31 +918,44 @@ realpath(const char *path, char resolved[PATH_MAX])
 	return resolved;
 }
 
+/* This function is not thread safe. 
+* TODO - It uses static memory. Is this a good design?
+*/
 char*
-sanitized_path(const char *path)
+resolved_path(const char *input_path)
 {
-	if(!path) return NULL;
-
+	static char resolved_path[PATH_MAX] = {0,};
 	static char newPath[PATH_MAX] = { '\0', };
 	errno_t r = 0;
 
-	if (path[0] == '/' && path[1]) {
-		if (path[2] == ':') {
-			if (path[3] == '\0') { /* make "/x:" as "x:\\" */
-				if((r = strncpy_s(newPath, sizeof(newPath), path + 1, strlen(path) - 1)) != 0 ) {
-					debug3("memcpy_s failed with error: %d.", r);
-					return NULL;
-				}
-				newPath[2] = '\\';
-				newPath[3] = '\0';
-
-				return newPath;
+	if (!input_path) return NULL;
+
+	/* If filename contains __PROGRAMDATA__ then expand it to %programData% and return the resolved path */
+	if (NULL != strstr(input_path, PROGRAM_DATA)) {
+		resolved_path[0] = '\0';
+		strcat_s(resolved_path, _countof(resolved_path), get_program_data_path());
+		strcat_s(resolved_path, _countof(resolved_path), &input_path[strlen(PROGRAM_DATA)]);
+
+		return resolved_path; /* return here as its doesn't start with "/" */
+	}
+
+	strcpy_s(resolved_path, _countof(resolved_path), input_path);
```

Comment:
what happens if this breaches MAX_PATH ?

Reply:
All our windows code relies on PATH_MAX.. If we breach PATH_MAX then we run into functionality bugs.

Reply:
I understand. My question if there any security implications of such breaches (crashes, overruns) ?

Reply:
we are using secured APIs strcat_s, strcpy_s that takes the size of destination buffer so if the input buffer is more then we will copy till the size of destination buffer.. so there will be no crashes/overruns?